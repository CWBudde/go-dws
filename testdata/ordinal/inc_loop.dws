// Test Inc() in loop context
// Task 9.24: Inc() built-in function

var sum: Integer := 0;
var i: Integer := 0;

while i < 5 do
begin
	sum := sum + i;
	Inc(i);
end;

PrintLn('Sum: ', sum);
PrintLn('Final i: ', i);

// Test Inc with delta in loop
var count: Integer := 0;
var j: Integer := 0;

while j < 20 do
begin
	Inc(count);
	Inc(j, 3);
end;

PrintLn('Count: ', count);
PrintLn('Final j: ', j);
