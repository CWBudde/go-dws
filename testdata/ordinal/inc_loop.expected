Sum:  10
Final i:  5
Count:  7
Final j:  21
